packages/credential-eip712/src/agent/CredentialEIP712.ts

```diff
@@ -0,0 +1,321 @@
+import {
+  IAgentPlugin,
+  VerifiableCredential,
+  CredentialPayload,
+  VerifiablePresentation,
+  PresentationPayload,
+  IIdentifier,
+} from "@veramo/core"
+import {
+  _ExtendedIKey,
+  extractIssuer,
+  MANDATORY_CREDENTIAL_CONTEXT,
+  processEntryToArray,
+  resolveDidOrThrow,
+  mapIdentifierKeysToDoc,
+  getChainIdForDidEthr,
+  getEthereumAddress,
+  isDefined
+} from "@veramo/utils"
+import { schema } from '../index'
+
+import {
+  ICreateVerifiableCredentialEIP712Args,
+  ICreateVerifiablePresentationEIP712Args,
+  ICredentialIssuerEIP712,
+  IRequiredContext,
+  IVerifyCredentialEIP712Args,
+  IVerifyPresentationEIP712Args,
+} from '../types/ICredentialEIP712'
+import { recoverTypedSignature, normalize, SignTypedDataVersion } from '@metamask/eth-sig-util'
+
+import { getEthTypesFromInputDoc } from "eip-712-types-generation"
+
+/**
+ * A Veramo plugin that implements the {@link ICredentialIssuerEIP712} methods.
+ *
+ * @public
+ */
+export class CredentialIssuerEIP712 implements IAgentPlugin {
+  readonly methods: ICredentialIssuerEIP712
+  readonly schema = schema.ICredentialIssuer
```

Comment:
I think there's a typo here:
```suggestion
  readonly schema = schema.ICredentialIssuerEIP712
```